nixos/modules/services/x11/xserver.nix

```diff
@@ -313,12 +335,27 @@ in
 
       xrandrHeads = mkOption {
         default = [];
-        example = [ "HDMI-0" "DVI-0" ];
-        type = with types; listOf string;
+        example = [ "HDMI-0" { output = "DVI-0"; primary = true; monitorConfig = ""; } ];
+        type = with types; listOf (either
+          str
+          (submodule {
+            options.output = str;
+            options.primary = bool;
+            options.monitorConfig = lines;
```

Comment:
Did you know it works without having `mkOption`?
On 16/03/2016 4:45 AM, "aszlig" <notifications@github.com> wrote:

> In nixos/modules/services/x11/xserver.nix
> <https://github.com/NixOS/nixpkgs/pull/13916#discussion_r56208116>:
>
> > @@ -313,12 +335,27 @@ in
> >
> >        xrandrHeads = mkOption {
> >          default = [];
> > -        example = [ "HDMI-0" "DVI-0" ];
> > -        type = with types; listOf string;
> > +        example = [ "HDMI-0" { output = "DVI-0"; primary = true; monitorConfig = ""; } ];
> > +        type = with types; listOf (either
> > +          str
> > +          (submodule {
> > +            options.output = str;
> > +            options.primary = bool;
> > +            options.monitorConfig = lines;
>
> These need to be valid options (-> mkOption) as well, like:
>
> submodule {
>   options.output = mkOption {
>     type = types.str;
>     description = ''      The display connector used for this XRandR head.    '';
>   };
> }
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/13916/files/47535fdc8c0eca8fd4b26f2601c602655f41a741#r56208116>
>